client/src/components/Ticketing/ticketingmanager/AdminPurchase/AdminPurchase.tsx

```diff
@@ -317,81 +98,91 @@ const AdminPurchase = () => {
 
     const updatedRows = eventData.map((r) => {
       if (r.id === row.id) {
+        delete r.ticketTypes;
+        delete r.seatsForType;
+        delete r.typeID;
         return {
           ...row,
           ...matchingEvent,
           eventtime: null,
           eventinstanceid: null,
+          ticketRestrictionInfo: [initialTicketTypeRestriction],
         };
       }
       return r;
     });
+    setPriceByRowId((prevState) => ({
+      ...prevState,
+      [row.id]: 0,
+    }));
     setEventData(updatedRows);
   };
 
-  const handleTimeChange = (
+  const handleTimeChange = async (
     event: React.ChangeEvent<HTMLSelectElement>,
     row: EventRow,
   ) => {
     const eventInstanceID = parseInt(event.target.value);
-    const selectedEvent = availableTimesByRowId[row.id]?.find(
+    const selectedEventInstance = availableTimesByRowId[row.id]?.find(
       (e) => e.eventinstanceid === eventInstanceID,
     );
+
+    // get ticket restrictions for event instance
+    const eventInstanceTicketRestrictions = allTicketRestrictions.filter(
+      (restriction) => {
+        return restriction.eventinstanceid === eventInstanceID;
+      },
+    );
+
+    const ticketRestictionWithDescription = eventInstanceTicketRestrictions.map(
+      (restriction) => {
+        const matchingTicketType = allTicketTypes.find(
+          (type) => type.tickettypeid === restriction.tickettypeid,
+        );
+
+        const {description} = matchingTicketType;
+
+        return {...restriction, tickettypedescription: description};
+      },
+    );
```

Comment:
I don't think this is necessary as the `ticket-restriction` API returns the description with the rest of the info